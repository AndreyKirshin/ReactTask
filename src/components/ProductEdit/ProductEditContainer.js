import React, { Fragment, useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Header from '../Header/Header';
import { fetchCategories } from '../../actions/categories';
import { fetchProducts, editProduct } from '../../actions/products';
import { getProductById } from '../../reducers/products';
import ProductForm from '../ProductForm/ProductForm';
import 'react-notifications/lib/notifications.css';
import {NotificationContainer, NotificationManager} from 'react-notifications';

export const ProductEditContainer = ({
  categories,
  getCaregories,
  getProducts,
  editProduct,
  notifications,
  product
}) => {

  useEffect(() => {
    getCaregories();
    getProducts();
  }, [getCaregories, getProducts]);

  
  useEffect(() => {
    if (notifications) {
      NotificationManager.success('Product has been updated', 'Success!');
    }
  }, [notifications]);
  
  
  return (
    <Fragment>
      <Header name="Edit Product"/>
      <div>we will edit here</div>
      <ProductForm categories={categories} sendProduct={editProduct} product={product}/>
      <NotificationContainer/>
    </Fragment>
  )
}

ProductEditContainer.propTypes = {
  categories: PropTypes.array.isRequired,
  getCaregories: PropTypes.func.isRequired,
  getProducts: PropTypes.func.isRequired,
  editProduct: PropTypes.func.isRequired,
  notifications: PropTypes.bool.isRequired,
  product: PropTypes.object
};

ProductEditContainer.defaultProps = {
  product: null
};

const mapStateToProps = state => {
  const { categories, notifications } = state;
  const product = getProductById(state)(window.location.pathname.split('/').pop());

  return {
    categories,
    notifications,
    product
  }
};

const mapDispatchToProps = dispatch => ({
  getCaregories: () => dispatch(fetchCategories()),
  getProducts: () => dispatch(fetchProducts()),
  editProduct: product => dispatch(editProduct(product))
});

export default connect(mapStateToProps, mapDispatchToProps)(ProductEditContainer);