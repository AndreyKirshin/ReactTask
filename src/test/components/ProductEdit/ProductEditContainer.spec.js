import React from 'react';
import { ProductEditContainer } from '../../../components/ProductEdit/ProductEditContainer';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


describe('<ProductEditContainer/>', () => {
  const props = {
    categories: [],
    getCaregories: jest.fn(),
    getProducts: jest.fn(),
    editProduct: jest.fn(),
    notifications: false,
    product: {
      name: '111',
      id: 1,
      categories: [1, 2, 3],
      rating: 3
    }
  }
  it('should match snapchot', () => {
    const wrapper = shallow(<ProductEditContainer {...props}/>);

    expect(wrapper).toMatchSnapshot()
  })
})