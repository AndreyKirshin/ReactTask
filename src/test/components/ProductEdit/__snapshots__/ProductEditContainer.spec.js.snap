// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductEditContainer/> should match snapchot 1`] = `ShallowWrapper {}`;
